tests/modules/images/webp-uploads/load-tests.php

```diff
@@ -775,6 +775,7 @@ public function it_should_create_webp_for_full_size_which_is_smaller_in_webp_for
 		$this->opt_in_to_jpeg_and_webp();
 
 		add_filter( 'webp_uploads_discard_larger_generated_images', '__return_true' );
+		remove_all_filters( 'wp_editor_set_quality' );
```

Comment:
Why is the new filter being removed here?

Reply:
New image quality for WebP is actually generating smaller image size then jpeg image in this test case. So we are testing it if full image has webp and other sizes doesn't have webp image in sources. But with new quality, it generates sources for all image sizes. so this case fails. That is why I have removed the quality filter from here as well as from the test case below.

Reply:
Should this perhaps be removed on a per-test basis instead? It seems odd that we would add a filter as part of the module, only to disable it for all of the module's tests. Is that not the case here?

Reply:
@aaemnnosttv Isn't the filter already being removed on a per-test basis here? I think that looks reasonable to me.

However: @mehulkaklotar Instead of relying on whatever quality is set by default, I think we should rather solve this in a more reliable way: Instead of removing the filter and relying on whatever WP core provides, why don't we add a custom filter in these tests instead, which sets the quality for all `image/webp` MIME type images to e.g. 90? That should ensure that no matter what core's behavior is, the WebP images generated here will indeed be larger.

Reply:
> Isn't the filter already being removed on a per-test basis here? I think that looks reasonable to me.

Ah yes, my mistake – I had misread the context of the diff 👍 

> Instead of removing the filter and relying on whatever WP core provides, why don't we add a custom filter in these tests instead, which sets the quality for all `image/webp` MIME type images to e.g. 90? That should ensure that no matter what core's behavior is, the WebP images generated here will indeed be larger.

This sounds better to me as well, although should we perhaps also filter the quality for jpg images to be lower as well to make the difference explicit and potentially even larger?

Reply:
I have added a new filter for these tests to return WebP quality to 86. Actually with 90, it was still failing. so I am just sticking with 86 for WebP in these tests.